co-simulation tool/docker/ns3-docker/user-guide-docker.md

```diff
@@ -0,0 +1,222 @@
+#  Running CARLA-SUMO-MOSAIC Co-Simulation Tool with NS-3 Integration in Docker 
+
+This is the tutorial to run CARLA-SUMO-MOSAIC co-simulation tool in a docker container. It is recommended to run the co-simulation tool on a **high-performance** computer.
+
+## Test Platform
+
+- Operating System: Ubuntu 20.04.2 LTS 
+- CPU: Intel(R) Core(TM) i7-9700K CPU @ 3.60 GHz
+- RAM: 16 GB
+- Graphics: NVIDIA GeForce RTX 2070 Super 8 GB
+
+## Prerequisites
+
+- NVIDIA GPU with at least 8 GB
+- NVIDIA drivers > 361.93
+- Free Disk Space > 30 GB
+- NVIDIA Container Toolkit
+- Docker
+
+If you do not have Docker installed, please refer to this [link](https://docs.docker.com/engine/install/ubuntu/) for installation.
+
+If you do not have NVIDIA Container Toolkit installed, please refer to this [link](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html) for installation.
+
+## Build Docker Image of Co-simulation Tool
+
+#### Preparation for building docker image
+
+- Put the docker file (Dockerfile), the folder of CARLA executable file (CARLA_0.9.10), and the folder of `ns-3-integration`in the same folder.
+- Change the CARLA path in the CARLA configuration file of scenarios to the location of the CARLA executable file.
+- The .sh files (CarlaUE4.sh and bridge.sh) must be given permission.
+
+#### Build docker image
+
+##### Step 1:  Run the following command to build the docker image
+
+```
+docker build - < Dockerfile -t ns3-integration
+```
+
+##### Step 2: Copy CARLA and Co-Simulation tool with NS-3 to the docker image and commit them
+
+Copy `CARLA_0.9.10` and `ns-3-integration` folder into this container.
+
+```
+docker cp CARLA_0.9.10 <container-id>:./
+docker cp ns-3-integration <container-id>:./
+docker container commit <container-id> ns-3-integration
+```
+
+### Run Docker Image
+
+```
+docker run --rm -it --gpus all -v /tmp/.X11-unix:/tmp/.X11-unix -e DISPLAY=$DISPLAY -e QT_X11_NO_MITSHM=1 --user=carla_sumo_mosaic ns-3-integration
+```
+
+### Modify files in Docker Container
+
+#### Step 1: Modify `ns3_installer.sh`
+
+In Docker container, type:
+
+```
+cd ns-3-integration/bin/fed/ns3
+```
+
+Then modify `ns3_installer.sh` by:
+
+```
+vim ns3_installer.sh
+```
+
+(If you see a message `vim: command not found`, please use `sudo apt-get install vim` to install vim as text editor)
+
+In line 365, change
```

Comment:
It seems like not in the exactly line 365 from my side. Could you please double check about this part?

Reply:
The location has been specified in the user guide.